### PR TITLE
TAJO-1768: Move Tajo into Java 8

### PR DESCRIPTION
**Pull request by hyunsik**
[Review on Reviewable](https://reviewable.io/reviews/apache/tajo/692)

### CODE REVIEW

**Comment by jinossy** [comment]
Can you trigger travis CI? and supporting jdk should be announced to mailing list

**Comment by hyunsik** [comment]
It is ready to be reviewed.

File: tajo-core/src/main/java/org/apache/tajo/engine/planner/Projector.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
 import org.apache.tajo.SessionVars;
 import org.apache.tajo.catalog.Schema;
+import org.apache.tajo.datum.Datum;
```
Comment (jihoonson) [inline tajo-core/src/main/java/org/apache/tajo/engine/planner/Projector.java]: Unused import

File: tajo-plan/src/main/java/org/apache/tajo/plan/joinorder/GreedyHeuristicJoinOrderAlgorithm.java [hunk 2]
```diff
@@ -484,7 +487,7 @@ public static double getCost(LogicalNode node) {
     default:
       // all binary operators (join, union, except, and intersect) are handled in the above cases.
       // So, we need to handle only unary nodes in default.
-      cost = getCost(((UnaryNode) node).getChild());
+      cost = getCost((LogicalNode)((UnaryNode) node).getChild());
```
Comment (jihoonson) [inline tajo-plan/src/main/java/org/apache/tajo/plan/joinorder/GreedyHeuristicJoinOrderAlgorithm.java]: Duplicate cast

**Comment by jihoonson** [comment]
I'm really waiting for this patch! I left very trivial comments.
In addition, you need to change the jdk version in BUILDING, README, and getting_started.rst.

**Comment by hyunsik** [comment]
Thank you for your comments.

File: tajo-plan/src/main/java/org/apache/tajo/plan/joinorder/GreedyHeuristicJoinOrderAlgorithm.java (hunk 2 above)
Comment (hyunsik) [inline tajo-plan/src/main/java/org/apache/tajo/plan/joinorder/GreedyHeuristicJoinOrderAlgorithm.java]: Basically, It cannot be avoidable. The problem is that both getChild() and getCost() require type inference.

**Comment by hyunsik** [comment]
I rebased and reflected your comment.

**Comment by jihoonson** [comment]
+1 ship it!

**Pull request closed without merging**